transport/outbound.go

```diff
@@ -27,6 +27,12 @@ import "golang.org/x/net/context"
 // Outbound is a transport that knows how to send requests for procedure
 // calls.
 type Outbound interface {
+	// Sets up the outbound to start making calls.
```

Comment:
can you make a note that `Call` cannot be used until `Start` has returned successfully?

should have outbounds verify this and return an error if `Start` hasn't been called yet?